### ISSUE TITLE
caching

### ISSUE THREAD

**Issue by charcoalberry** [silveredgold/beta-censoring#22]
It seems like the server spends a lot of time analyzing the same images again and again. I understand that caching is a subject with many pitfalls, but wouldn't it be beneficial to at least keep a list of urls (or hashes of downloaded files) that were examined but not censored, to avoid running those through the neural network again? That way the server could skip most of the fluff of websites, and focus on the images that matter.

**Comment by silveredgold** [comment]
Yeah I'd always put off caching since, as you mentioned, it does tend to run into a few pitfalls. I'll look into at least adding some options so that people can opt-in to some form of caching maybe.

When looking into it previously, the performance improvements weren't that dramatic but that was also back in the prototyping stage so it might make more sense now.

**Comment by mailnmailn** [comment]
+1 for caching, keeping hashes of non-censored images would seem like the way to go to me. Maybe also a filter for images below certain dimensions in px?

Currently, the server is extremely slow when browsing reddit, for instance, as it censors all the UI elements and background/sidebar images on every new page. This often represents 2/3rds or more of the images censored when loading a new page, and makes it much slower than it could be

**Comment by silveredgold** [comment]
Caching is now available and (partially) enabled by default in v0.0.11. Docs for configuring caching are available [here](https://silveredgold.github.io/beta-censoring/content/configuration.html#caching-configuration)
